### PR TITLE
Refactor Tutorials, Guides, and Intro in MLflow Docs

### PR DESCRIPTION
**Pull request by BenWilson2**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

- New introduction to the Docs
- Template for tutorials with first getting started tutorial
- some new css elements for within-docs formatting / visual organization structures

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

## Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [X] Yes. I've updated:
  - [X] Examples
  - [ ] API references
  - [X] Instructions

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by BenWilson2** [comment]
Still very much a WIP - pushing draft for style reference for others.

File: docs/source/_static/images/tutorials/introductory/logging-first-model/default-experiment.gif
Comment (harupy) [inline docs/source/_static/images/tutorials/introductory/logging-first-model/default-experiment.gif:1]: Do we need this GIT? Can we just provide the notebook with outputs?

File: docs/source/_static/images/tutorials/introductory/logging-first-model/training-annotation.svg
Comment (harupy) [inline docs/source/_static/images/tutorials/introductory/logging-first-model/training-annotation.svg:1]: Can we use code and comments?

File: docs/source/tutorials/introductory/logging-first-model/step6-logging-a-run.rst [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+            params = {
+                "n_estimators": 100,
+                "max_depth": 6,
+                "min_samples_split": 10,
+                "min_samples_leaf": 4,
+                "bootstrap": True,
+                "oob_score": False,
+                "random_state": 888
+            }
```
Comment (harupy) [inline docs/source/tutorials/introductory/logging-first-model/step6-logging-a-run.rst:149]: I'd reorganize the code like this:

```python
params = ...
metrics = ...
model = ...

with mlflow.start_run(run_name=run_name) as run:
    mlflow.log_params(params)
    mlflow.log_metrics(metrics)
    mlflow.sklearn.log_model(
        sk_model=rf,
        input_example=X_val,
        artifact_path=artifact_path
    )
```

File: docs/source/tutorials/introductory/logging-first-model/step1-tracking-server.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+The first thing that we need to do is to get MLflow.
+
+Step 1: Install MLflow from PyPI
+--------------------------------
+
+MLflow is conveniently available on PyPI. Installing it is as simple as running a pip command.
+
+.. code-section::
+    .. code-block:: bash
+        :caption: Install the Latest MLflow Version from PyPI
```
Comment (mberk06) [inline docs/source/tutorials/introductory/logging-first-model/step1-tracking-server.rst:19]: Potentially controversial but I think all captions strongly decrease readability. IMO we should change caption rendering to the gif "italics under the block" style or remove.

Comment (BenWilson2) [reply docs/source/tutorials/introductory/logging-first-model/step1-tracking-server.rst]: not controversial AT ALL. Removed them all :D They were super distracting!

File: docs/source/tutorials/introductory/logging-first-model/step6-logging-a-run.rst (hunk 1 above)
Comment (BenWilson2) [reply docs/source/tutorials/introductory/logging-first-model/step6-logging-a-run.rst]: good idea. Changed in all places (docs and notebooks)

File: docs/source/index.rst [hunk 3]
```diff
@@ -1,28 +1,340 @@
...
-Get started using the :ref:`quickstart` or by reading about the :ref:`key concepts<concepts>`.
 
 .. toctree::
     :maxdepth: 1
+    :hidden:
 
-    what-is-mlflow
+    introduction/index
+    tutorials/index
+    guides/index
```
Comment (dbczumar) [inline docs/source/index.rst:337]: @BenWilson2 Love what we're doing in this PR. The CSS & embedded notebooks are awesome!

One thing I'd like to adjust / align on is the TOC tree. A couple things I've noticed:

1. We seem to have Quickstarts (https://mlflow.org/docs/latest/quickstart.html), Tutorials, and Guides. The distinction between each is not clear.

2. All of the tutorials are colocated in the same section, which will make it harder to organize things.

**Proposal**: Rather than having Quickstarts, Tutorials, and Guides at the top-level, I propose we structure the TOC tree into topics. Each topic has an overview, guide, and tutorial.

- MLflow Overview

- Getting Started
   - Quickstarts

- LLMs
   - Guide
   - Tutorials
  
- Classical ML
   - Guide
   - Tutorials

- Deep Learning
   - Guide
   - Tutorials

- Production Model Management

- Deployment
   - Guide
   - Tutorials

...

WDYT?

File: docs/source/introduction/index.rst [hunk 4]
```diff
@@ -0,0 +1,163 @@
+MLflow 101
```
Comment (dbczumar) [inline docs/source/introduction/index.rst:1]: Can we call this `What is MLflow?`. This is better for SEO and friendlier for non-english speakers.

I.e. instead of removing the existing page, can we update it with the content you've written here?

File: docs/source/guides/index.rst [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+---------------
+
+For those looking to implement MLflow in specific scenarios or wanting to learn best practices
+for unique use cases, these guides are invaluable. Like our tutorials, every guide comes with
+supporting materials, be it code snippets, notebooks, or scripts, so you can get hands-on
+experience.
+
+.. raw:: html
+
+    <section class="card-list">
```
Comment (harupy) [inline docs/source/guides/index.rst:23]: can we start without cards?

File: docs/source/_static/images/guides/introductory/hyperparameter-tuning-with-child-runs-notebook.gif
Comment (harupy) [inline docs/source/_static/images/guides/introductory/hyperparameter-tuning-with-child-runs-notebook.gif:1]: can we minimize the number of GIFs? They are not copiable and require more effort to update than text.

File: docs/source/tutorials/introductory/logging-first-model/step5-synthetic-data.rst [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+.. code-section::
+    .. code-block:: python
+
+        import pandas as pd
+        import numpy as np
+        from datetime import datetime, timedelta
+
+
+        def generate_apple_sales_data_with_promo_adjustment(
```
Comment (harupy) [inline docs/source/tutorials/introductory/logging-first-model/step5-synthetic-data.rst:31]: Can we use the iris dataset?

Comment (BenWilson2) [reply docs/source/tutorials/introductory/logging-first-model/step5-synthetic-data.rst:31]: The iris data set doesn't work well due to the propensity for models to 'memorize' the features, reducing the utility of generating realistic plots, loss values, and makes hyperparameter tuning use cases pointless. We need a dataset that is free, has a distinct amount of covariance, has varying degrees of causal interactivity between the features and the target and among the features themselves in order to meet the needs of both the tutorial and the guide for hyperparameter tuning. 
Datasets of sufficient size that exhibit these traits are either not free to use (UCI datasets with restrictive licenses) or have significant issues with their content (missing / malformed data) or require fetching from a location like kaggle (which is not openly free to download from without an account). 
Because of all of these reasons, a hand-crafted synthetic dataset is the best approach to meet all of the demonstration and tutorial needs.

Comment (harupy) [reply docs/source/tutorials/introductory/logging-first-model/step5-synthetic-data.rst:31]: I think it's fine to generate unrealistic or pointless plots/models as long as we can demonstrate what MLlfow is capable of.

File: docs/source/_static/images/guides/introductory/hyperparameter-tuning-with-child-runs-notebook.gif
Comment (WeichenXu123) [reply docs/source/_static/images/guides/introductory/hyperparameter-tuning-with-child-runs-notebook.gif:1]: +1, maybe we can replace them with picture link on internet

File: docs/source/introduction/index.rst (hunk 4 above)
Comment (WeichenXu123) [reply docs/source/introduction/index.rst:1]: For SEO purpose, shall we also add meta tags ? https://docs.readthedocs.io/en/stable/guides/technical-docs-seo-guide.html#use-meta-tags It helps search engine displays the website abstract better.

File: docs/source/index.rst (hunk 3 above)
Comment (WeichenXu123) [reply docs/source/index.rst:337]: Looks good to me :)

File: docs/source/guides/introductory/hyperparameter-tuning-with-child-runs/part2-logging-plots.rst [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+
+In this portion of the guide, we look at a notebook that is concerned with the generation and storage of
+common and useful plots associated with a regression task.
+
+We'll be looking at two primary means of logging plots along with our logged models:
+
+- **Direct plot logging** via ``mlflow.log_figure()`` we will use an in-memory figure reference to a generated plot.
+- **Logging a local plot file** via ``mlflow.log_artifact()`` to allow us to log a locally stored image to the run.
+
+Role of Visualizations in Model Analysis
```
Comment (harupy) [inline docs/source/guides/introductory/hyperparameter-tuning-with-child-runs/part2-logging-plots.rst:19]: Is this something we need to explain?

File: docs/source/guides/introductory/hyperparameter-tuning-with-child-runs/part2-logging-plots.rst [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+especially in a notebook environment, to ensure that each training iteration has a specific and
+unambiguous reference to the plot generation, directly linked to the exact state of the data used
+in a training iteration. Such a method mitigates the risks associated with declaratively defined and
+materialized plots, which, if not regenerated after data modification, can lead to inconsistencies
+and errors in data representation.
+
+.. code-section::
+    .. code-block:: python
+
+        def plot_box_weekend(df, style="seaborn", plot_size=(10, 8)):
```
Comment (harupy) [inline docs/source/guides/introductory/hyperparameter-tuning-with-child-runs/part2-logging-plots.rst:110]: Can we log a toy plot? 

```python
import mlflow
import matplotlib.pyplot as plt

fig, ax = plt.subplots()
ax.plot([0, 1], [2, 3])

with mlflow.start_run():
    mlflow.log_figure(fig, "figure.png")
```

File: docs/source/tutorials/introductory/logging-first-model/step5-synthetic-data.rst (hunk 6 above)
Comment (harupy) [reply docs/source/tutorials/introductory/logging-first-model/step5-synthetic-data.rst:31]: It doesn't have to be iris. Any simple dataset is ok.

File: docs/source/introduction/index.rst [hunk 9]
```diff
@@ -0,0 +1,144 @@
+What is Mlflow?
```
Comment (harupy) [inline docs/source/introduction/index.rst:1]:
```suggestion
What is MLflow?
```

File: docs/source/llms/index.rst [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+
+Such complexities can easily become a bottleneck for developers and data scientists aiming to integrate LLM 
+capabilities into their applications.
+
+**MLflow's Support for LLMs** aims to alleviate these challenges by introducing a suite of features and tools designed with the end-user in mind:
+
+.. raw:: html
+
+    <div class="redirect-header">
+        <a href="gateway/index.html">⛩️ MLflow AI Gateway ⛩️</a>
```
Comment (harupy) [inline docs/source/llms/index.rst:17]:
```suggestion
        <a href="gateway/index.html">MLflow AI Gateway</a>
```

**Comment by harupy** [comment]
Can we split this PR? I think it's too large and difficult to review.

File: docs/source/_static/images/tutorials/introductory/logging-first-model/creating-experiment.gif
Comment (harupy) [inline docs/source/_static/images/tutorials/introductory/logging-first-model/creating-experiment.gif:1]: do we need this GIF?

File: docs/source/_static/images/tutorials/introductory/logging-first-model/default-experiment.gif
Comment (harupy) [inline docs/source/_static/images/tutorials/introductory/logging-first-model/default-experiment.gif:1]: do we need this GIF?

File: docs/source/getting-started/logging-first-model/step6-logging-a-run.rst [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+                sk_model=rf, input_example=X_val, artifact_path=artifact_path
+            )
+
+
+To aid in visualizing how MLflow tracking API calls add in to an ML training code base, see the figure below.
+
+.. figure:: ../../_static/images/tutorials/introductory/logging-first-model/training-annotation.svg
+   :width: 1024px
+   :align: center
+   :alt: Explanation of MLflow integration into ML training code
```
Comment (harupy) [inline docs/source/getting-started/logging-first-model/step6-logging-a-run.rst:182]: Do we need this part?

File: docs/source/traditional-ml/index.rst [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+                    <img src="../_static/images/logos/statsmodels-logo.svg" alt="Statsmodels Logo" class="logo">
+                </a>
+            </div>
+            <div class="logo-box">
+                <a href="../models.html#prophet-prophet">
+                    <img src="../_static/images/logos/prophet-logo.png" alt="Prophet Logo" class="logo">
+                </a>
+            </div>
+        </div>
+    </section>
```
Comment (harupy) [inline docs/source/traditional-ml/index.rst:69]: Do we need these logos? Can we just use a list of links?

File: docs/source/llms/index.rst (hunk 10 above)
Comment (BenWilson2) [reply docs/source/llms/index.rst:17]: The new design will be replacing these placeholder elements

File: docs/source/traditional-ml/index.rst (hunk 12 above)
Comment (BenWilson2) [reply docs/source/traditional-ml/index.rst:69]: This placeholder is getting redesigned (it will be logo cards that Ridhima is going to design for us)
